src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -825,19 +842,65 @@ namespace AppInstaller::CLI::Workflow
         const Manifest::ManifestInstaller& first,
         const Manifest::ManifestInstaller& second)
     {
+        // The priority will still be used as a tie-break between weak results.
+        std::optional<std::string_view> firstWeakComparator;
+        bool firstWeakComparatorResult = false;
+
         for (auto comparator : m_comparators)
         {
-            if (comparator->IsFirstBetter(first, second))
+            details::ComparisonResult forwardCompare = comparator->IsFirstBetter(first, second);
+
+            if (forwardCompare == details::ComparisonResult::StrongPositive)
             {
-                AICLI_LOG(CLI, Verbose, << "Installer " << first << " is better than " << second << " due to: " << comparator->Name());
+                AICLI_LOG(CLI, Verbose, << "Installer " << first << " is better [strong] than " << second << " due to: " << comparator->Name());
                 return true;
             }
-            else if (comparator->IsFirstBetter(second, first))
+
+            details::ComparisonResult backwardCompare = comparator->IsFirstBetter(second, first);
+
+            if (backwardCompare == details::ComparisonResult::StrongPositive)
             {
                 // Second is better by this comparator, don't allow a lower priority one to override that.
-                AICLI_LOG(CLI, Verbose, << "Installer " << second << " is better than " << first << " due to: " << comparator->Name());
+                AICLI_LOG(CLI, Verbose, << "Installer " << second << " is better [strong] than " << first << " due to: " << comparator->Name());
                 return false;
             }
+
+            // Save the first weak result that we get
+            if (!firstWeakComparator)
+            {
+                if (forwardCompare == details::ComparisonResult::WeakPositive)
+                {
+                    firstWeakComparator = comparator->Name();
+                    firstWeakComparatorResult = true;
+                }
+                else if (backwardCompare == details::ComparisonResult::WeakPositive)
+                {
+                    firstWeakComparator = comparator->Name();
+                    firstWeakComparatorResult = false;
+                }
+            }
+
+            // Should not happen, but if it does it points at a serious bug that should be fixed.
+            if (forwardCompare == details::ComparisonResult::WeakPositive && backwardCompare == details::ComparisonResult::WeakPositive)

```

Comment:
Would we gain anything by also checking for both being strong positives? (at the top of the block)